### PR TITLE
SOC-965 Rename login/signup to signin/register

### PR DESCRIPTION
**Pull request by bkoval**
https://wikia-inc.atlassian.net/browse/SOC-965 @kvas-damian @mklucsarits

### CODE REVIEW

File: server/facets/auth/register.ts [hunk 1]
```diff
@@ -4,11 +4,13 @@
 
 import BirthdateInput = require('./BirthdateInput');
 import dateUtils = require('../../lib/DateUtils');
+import authUtils = require('../../lib/AuthUtils');
```
Comment (kvas-damian) [inline server/facets/auth/register.ts]: Why there are new requires? I don't see any usages.

File: server/lib/AuthUtils.ts [hunk 2]
```diff
@@ -38,3 +38,9 @@ export function getCacheBusterUrl(redirect: string): string {
 
 	return url.format(cacheBustedUrlObj);
 }
+
+export function getRedirectUrlWithQueryString(route: string, request: Hapi.Request) {
+	var redirectUrl = request.url;
+	redirectUrl.pathname = route;
+	return url.format(redirectUrl);
```
Comment (kvas-damian) [inline server/lib/AuthUtils.ts]: Isn't it more readable to use ```redirectUrl.format()```?

File: server/facets/auth/register.ts (hunk 1 above)
Comment (bkoval) [inline server/facets/auth/register.ts]: authUtils are used in new handlers :)

File: server/facets/auth/register.ts [hunk 3]
```diff
@@ -6,9 +6,10 @@ import BirthdateInput = require('./BirthdateInput');
 import dateUtils = require('../../lib/DateUtils');
 import localSettings = require('../../../config/localSettings');
 import authView = require('./authView');
-var deepExtend = require('deep-extend');
+var deepExtend = require('deep-extend'),
+	url = require('url');
```
Comment (kvas-damian) [inline server/facets/auth/register.ts]: And where are we using this one? :P

**Comment by kvas-damian** [comment]
LGTM

File: front/locales/de/auth.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
 {
   "common":{
     "server-error":"Es tut uns leid, bei der Verarbeitung deiner Anfrage ist ein Fehler aufgetreten. Bitte versuche es später noch einmal.",
-    "login-callout":"Hast du bereits ein Benutzerkonto?",
+    "signin-callout":"Hast du bereits ein Benutzerkonto?",
```
Comment (mklucsarits) [inline front/locales/de/auth.json]: Why change these?

**Comment by mklucsarits** [comment]
I don't understand the necessity and don't personally approve of renaming "login" and "signup" to "signin" and "register" in the codebase. What is the reason for doing so? Is there a reason beyond the creation of the /signin and /register paths? Renaming "signup" to "register" is alright I guess, but pragmatically speaking, if someone is looking for log-in code in our codebase, no one is going to think to search for "signin" first. This will only lead to confusion. I also don't believe changing our log-in and registration paths necessitates changing the naming of variables, etc. in the codebase. It's important that we name classes and variables that make sense to us as developers, regardless of whether they match the words we show to our end-users.

**Comment by bkoval** [comment]
Main reason is consistency - since we're renaming the routes to reflect our product mission in a more elegant way and we want to be consistent in language everywhere in the UI, renaming of variables that correspond to route handlers, templates and context objects should make sense in this case.

I also think that if someone is going to be grepping the code, it's better to get all the results once the phrase "signin" is typed into a search form (especially that it corresponds to the route in the browser's address bar and to the product language), than get half / not full portion of results where a bit of them is referenced under signup-something and the others use login-something pattern.

Since routes, templates, i18n strings and context objects are tightly connected with user-facing views, I believe consistency is the biggest value that we should align with.

**Comment by mklucsarits** [comment]
But the naming was already consistent. And because we're keeping the /login and /signup paths, the new naming will be inconsistent with those. So we're not going to be able to have consistent naming between routes and variables/classes anyway, in either case.

File: front/locales/de/auth.json [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
     "login-link-text":"Anmelden",
```
Comment (mklucsarits) [inline front/locales/de/auth.json]: "login" is still used here. If you're renaming everything, be sure to grep and rename it all.

**Comment by bkoval** [comment]
We're keeping /login and /signup just to make them redirect to /signin and /register.

We can continue this discussion for a long time, but I believe variable naming (as long as it's consistent) is not worth discussing too much :)

**Comment by mklucsarits** [comment]
OK, I still don't think it's a good idea, but I won't hold up the merge.

**Pull request merged**